### PR TITLE
Desktop: enable device field for IRDA transports

### PR DESCRIPTION
**Pull request by bstoeger**
Fixes a bug introduced in d23bd46a1be2dfb25293639abcf06b5b4d4b94df:
device field was disabled for IRDA transports.

Signed-off-by: Berthold Stoeger <bstoeger@mail.tuwien.ac.at>

### Describe the pull request:

- [x] Bug fix
- [ ] Functional change
- [ ] New feature
- [ ] Code cleanup
- [ ] Build system change
- [ ] Documentation change
- [ ] Language translation

### CODE REVIEW

**Comment by janmulder** [comment]
Please get rid of the edited libdivecomputer "file". To do so, run a `git submodule update`. not sure you have to rebase after that (don't think so), and push again.

**Comment by bstoeger** [comment]
Gaah.. Doesn't work. I tried `git submodule update` in this branch and in my master branch. But it refuses to `rebase`:
> Current branch bt6 is up to date.

or to `push -f`:
> Everything up-to-date

I will have to redo this.

In the meanwhile, can you have a look at my resolution? I don't understand how this bug came to be. In the mentioned commit I only copied code, how did this ever work? Does the USB transport also need the device field?

**Comment by janmulder** [comment]
Yes, looked to the referenced commit, but that is indeed only some code reshuffle that does not explain the IRDA stuff being disabled. I was looking in a different direction (how is libdivecomputer compiled, with or without IRDA?), but before I got anywhere with that, your PR came in. And I'm unsure about IRDA. Not used that type of transport in maybe 10 years.

But I saw the question on our Google user group.

**Comment by bstoeger** [comment]
Wow - that was painful. The only way I found was to `formatpatch` and apply these patches manually. `cherry-pick` or `rebase -i` did not work. git really makes me feel stupid at times.

**Comment by bstoeger** [comment]
Something is definitely amiss here, which makes me doubt this resolution.

Note how `DC_TRANSPORT_BLUETOOTH` is defined to different values in `libdivecomputer/descriptor.h` and `desktop-widgets/downloadfromdivecomputer.cpp`. 

This cannot be healthy.

Unfortunately I will not be able to look into this tonight.

**Comment by janmulder** [comment]
@dirkhh: what about IRDA ... did this ever work? 

And what about that very weird commit f9cd09b00059 (@glance-) ? I reverted that to try and things still compile, and a BT download of my OSTC3 just works without that abusive re-define of DC_TRANSPORT_BLUETOOTH.

**Comment by bstoeger** [comment]
Yes, let's remove that redefinition - it *will* lead to crazy problems down the road.

**Comment by dirkhh** [comment]
Let's hold off on this for a moment and take a deep breath (I was busy fixing a crashed mail server anyway).
a) IRDA doesn't need a device, so it was correct to disable the input field
b) DC_TRANSPORT_BLUETOOTH -- the problem is that libdivecomputer didn't use to recognize BT as a transport - this is new code in libdivecomputer. I need to stare at our use of this a little longer, but I think we can simply use the enum from libdivecomputer again.

File: desktop-widgets/downloadfromdivecomputer.cpp [hunk 1]
```diff
@@ -480,13 +480,19 @@ void DownloadFromDCWidget::updateDeviceEnabled()
 	dc_descriptor_t *descriptor = NULL;
 	descriptor = descriptorLookup.value(ui.vendor->currentText() + ui.product->currentText());
 
-	// call dc_descriptor_get_transport to see if the dc_transport_t is DC_TRANSPORT_SERIAL
-	if (dc_descriptor_get_transport(descriptor) == DC_TRANSPORT_SERIAL) {
-		// if the dc_transport_t is DC_TRANSPORT_SERIAL, then enable the device node box.
+	// Enable the device selection field only for transports that need a device
+	switch (dc_descriptor_get_transport(descriptor)) {
+	case DC_TRANSPORT_SERIAL:
+	case DC_TRANSPORT_IRDA:
```
Comment (dirkhh) [inline desktop-widgets/downloadfromdivecomputer.cpp:486]: no, I think this is wrong

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,7 @@
 - Desktop: fix bug about progress bars not resizing
 - Desktop: fix a bug about dive list auto group when importing dives
 - Desktop: clear tags when creating a new logbook
+- Desktop: Enable device field for IRDA transports
```
Comment (dirkhh) [inline CHANGELOG.md:8]: nope, please don't

**Review by dirkhh — CHANGES_REQUESTED** [review]

**Comment by glance-** [comment]
@janmulder The old code in f9cd09b00059 was from the time where libdivecomputer had no clue what bluetooth was. It might be that define doesn't make sense any longer, but that should be synchronized with  the relevant update of libdivecomputer.

**Comment by bstoeger** [comment]
Alright - seems like the bug report was bogus. Let's close this.

The redefinition of `DC_TRANSPORT_BLUETOOTH` should be removed, but let's do this in a different PR.

**Pull request closed without merging**